### PR TITLE
[RLlib] td_lambda;reward bug;group loss;adam  Signed-off-by: wanc97 <wanc97@qq.com>

### PR DESCRIPTION
**Pull request by wanc97**
## Why are these changes needed?
- fix the bug of qmix in receiving reward;
- Add the target calculation method of td_lambda;  
- adjust the calculation method of loss to be group instead of agent;
- add adam

### CODE REVIEW

File: rllib/algorithms/qmix/qmix_policy.py [hunk 1]
```diff
@@ -349,10 +362,9 @@ def learn_on_batch(self, samples):
             next_action_mask,
             next_env_global_state,
         ) = self._unpack_observation(samples[SampleBatch.NEXT_OBS])
-        group_rewards = self._get_group_rewards(samples[SampleBatch.INFOS])
```
Comment (wanc97) [inline rllib/algorithms/qmix/qmix_policy.py:365]: rewards should come from rewards instead of infos

File: rllib/algorithms/qmix/qmix_policy.py [hunk 2]
```diff
@@ -96,61 +103,61 @@ def forward(
             mac_out, dim=3, index=actions.unsqueeze(3)
         ).squeeze(3)
 
-        # Calculate the Q-Values necessary for the target
-        target_mac_out = _unroll_mac(self.target_model, next_obs)
+        with torch.no_grad():
```
Comment (wanc97) [inline rllib/algorithms/qmix/qmix_policy.py:106]: Some gradient calculations do not require

**Comment by zhe-thoughts** [comment]
Hey @wanc97 thanks for the contribution! Ray team is busy with 2.2 releases and sorry about the delay in responding to your PR. 

cc @kouroshHakha , @gjoliver , @sven1977 for a review after we resolve other 2.2 related issues. Thanks

**Comment by avnishn** [comment]
hi @wanc97 sorry for the delays we'll take a look at this this week.

**Comment by wanc97** [comment]
> hi @wanc97 sorry for the delays we'll take a look at this this week.

Thank you, this commit comes from the problems I encountered in the process of using the qmix algorithm. After testing, almost 100% win rate can be achieved in most environments of StarCraft II.

**Comment by wanc97** [comment]
> 
Hello, excuse me, can you give some suggestions to this MR?
@avnishn @sven1977
